styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -191,7 +232,28 @@ private void updateStats(Map<String, Resource> resources,
 
   private boolean limitAndDequeue(Map<String, Resource> resources,
       Map<WorkflowId, Set<String>> workflowResourceReferences,
-      Map<String, Long> currentResourceUsage, InstanceState instance) {
+      Map<String, Long> currentResourceUsage, InstanceState instance,
+      CompletionStage<Optional<ExecutionBlocker>> executionBlockerFuture) {
+
+    // Check for execution blockers
+    Optional<ExecutionBlocker> blocker = Optional.empty();
+    try {
+      blocker = executionBlockerFuture.toCompletableFuture().get(30, TimeUnit.SECONDS);
```

Comment:
This would cause delay if somehow blocker execution is slow? e.g. bad network, external service slowness, corner cases, etc. Is 30s too much?

Reply:
What timeout would you prefer?

Note that the same is true where the scheduler calls `stateManager.receiveIgnoreClosed`. The scheduler might block for a long time for each workflow instance if the network is bad, the blocking datastore requests take a long time to complete, etc.

Reply:
Dunno :D How long would it require in our internal use case?
It is a little bit different comparing with communicating with datastore. The plugin can do arbitrary stuff and the platform has no control of that.

Reply:
I think `30` seconds is a reasonable timeout.

And I think it is fair that a scheduler plugin can cause problems for the scheduler. It can be disabled by using the `executionGatingEnabled` runtime config flag if necessary.

Reply:
Fair enough. A little bit documentation for `executionBlocker` method if you feel that's necessary?